### ISSUE TITLE
[google_sign_in] serverAuthCode is null on Android

### ISSUE THREAD

**Issue by Burchard36** [flutter/flutter#107816]
So, if you don't want to read through you can see my original post here on stack overflow: https://stackoverflow.com/questions/73017275/google-sign-in-flutter-plugin-always-returns-null-serveraccesscode/

But I will re-explain here:

When using the signIn() method and attempting to retrieve a `serverAuthCode`, no matter what was tried I always received a `null` return value, which I thought was odd as I'm switching over from java and this system was working perfectly over there with no additional setups (Eg the services.json, I'm on android btw) so I knew it couldn't of been my google setup.

So naturally, I went looking through many many post, with none of them helping my issues. However after many hours I finally found this: https://stackoverflow.com/questions/73017275/google-sign-in-flutter-plugin-always-returns-null-serveraccesscode/

So i decided to check the build.gradle for this plugin, and realized it shipped with:

`'com.google.android.gms:play-services-auth:20.0.5'`

While the latest version of this library is

`'com.google.android.gms:play-services-auth:20.2.0'`

After switching these imports, re-syncing and rebuild the serverAuthCode value was never null again

Im not entirely sure why this happened, maybe someone here has a better explanation? Here is the code i was using (Ill just send the whole file test it by setting the `home` in `main.dart` to `GoogleLoginPage()`)

<details>
<summary>Code sample</summary>

```dart
class GoogleLoginPage extends StatefulWidget {

  final String name = "Logging in with google.";

  late final GoogleSignIn _googleSignIn;

  GoogleLoginPage() : super() {
    _googleSignIn = GoogleSignIn(
        scopes: ['https://www.googleapis.com/auth/userinfo.email',
        'https://www.googleapis.com/auth/userinfo.profile'],
        clientId: "318189328718-7o68g6nmij39tssg0a4mgpkegmj3op07.apps.googleusercontent.com"
    );
  }

  Future<void> fetchAuthToken() async {
    try {
      var response = await _googleSignIn.signIn();
      log(response?.serverAuthCode ?? "No server auth code");
      _googleSignIn.signOut();
    } catch (error) {
      log("ERR");
      print(error);
    }
  }

  @override
  State<GoogleLoginPage> createState() => GoogleLoginState();

}

class GoogleLoginState extends State<GoogleLoginPage> {
  @override
  Widget build(BuildContext context) {
    // TODO: implement build
    return Scaffold(
      appBar: AppBar(
        title: Text(widget.name),
      ),
      body: Center(
        child: Text(widget.name),
      ),
      floatingActionButton: FloatingActionButton(
        onPressed: widget.fetchAuthToken,
        child: const Icon(Icons.add),
      ),
    );
  }
}
```

</details>

The call to signOut is just for my personal debugging purposes since I wanted the window to select an account each time I tested this. Hopefully someon else here is able to replicate my issue i had with serverAuthCode always being null on the current unmodified code-base.

If anymore information is needed please let me know. 

<details>
  <summary>Logs</summary>

`flutter run --verbose` output

I ran the signOn() method a few times you can see that at the end of the file
```
https://paste.md-5.net/ufeyehafat.bash
```

`flutter analyze` output
```
Analyzing civ_cracker...

   info - Don't import implementation files from another package - lib\generated_plugin_registrant.dart:10:8 -
          implementation_imports
   info - Use key in widget constructors - lib\google\google_login_page.dart:14:3 - use_key_in_widget_constructors
   info - Avoid `print` calls in production code - lib\google\google_login_page.dart:29:7 - avoid_print

3 issues found. (ran in 2.3s)
```

`flutter doctor -v` output
```
C:\Users\burch\civ_cracker>flutter doctor -v
[√] Flutter (Channel stable, 3.0.5, on Microsoft Windows [Version 10.0.19044.1826], locale en-US)
    • Flutter version 3.0.5 at C:\flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision f1875d570e (4 days ago), 2022-07-13 11:24:16 -0700
    • Engine revision e85ea0e79c
    • Dart version 2.17.6
    • DevTools version 2.12.2

[!] Android toolchain - develop for Android devices (Android SDK version 33.0.0)
    • Android SDK at C:\Users\burch\AppData\Local\Android\sdk
    X cmdline-tools component is missing
      Run `path/to/sdkmanager --install "cmdline-tools;latest"`
      See https://developer.android.com/studio/command-line for more details.
    X Android license status unknown.
      Run `flutter doctor --android-licenses` to accept the SDK licenses.
      See https://flutter.dev/docs/get-started/install/windows#android-setup for more details.

[√] Chrome - develop for the web
    • Chrome at C:\Program Files\Google\Chrome\Application\chrome.exe

[X] Visual Studio - develop for Windows
    X Visual Studio not installed; this is necessary for Windows development.
      Download at https://visualstudio.microsoft.com/downloads/.
      Please install the "Desktop development with C++" workload, including all of its default components

[√] Android Studio (version 2021.2)
    • Android Studio at C:\Program Files\Android\Android Studio
    • Flutter plugin can be installed from:
       https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
       https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.12+7-b1504.28-7817840)

[√] IntelliJ IDEA Community Edition (version 2022.1)
    • IntelliJ at C:\Program Files\JetBrains\IntelliJ IDEA Community Edition 2021.3
    • Flutter plugin version 69.0.4
    • Dart plugin version 221.5588

[√] VS Code (version 1.68.0)
    • VS Code at C:\Users\burch\AppData\Local\Programs\Microsoft VS Code
    • Flutter extension can be installed from:
       https://marketplace.visualstudio.com/items?itemName=Dart-Code.flutter

[√] Connected device (4 available)
    • moto g stylus (mobile) • ZY227HWN9K • android-arm64  • Android 10 (API 29)
    • Windows (desktop)      • windows    • windows-x64    • Microsoft Windows [Version 10.0.19044.1826]
    • Chrome (web)           • chrome     • web-javascript • Google Chrome 103.0.5060.114
    • Edge (web)             • edge       • web-javascript • Microsoft Edge 103.0.1264.62

[√] HTTP Host Availability
    • All required HTTP hosts are available

! Doctor found issues in 2 categories.
```

</details>

**Comment by maheshmnj** [comment]
Hi @Burchard36, Thanks for filing the issue. Please ensure you are using the latest version of the `google_sign_in`  plugin. I did verify on Android and it seems to return the serverAuthcode. 

<img width="790" alt="image" src="https://user-images.githubusercontent.com/31410839/179529582-3a9ba633-4135-4b4b-b709-8077507e5810.png">

This is a duplicate of a recently fixed issue https://github.com/flutter/flutter/issues/57712 (closed)

<details>
<summary>flutter doctor -v (mac)</summary>

```
[✓] Flutter (Channel stable, 3.0.5, on macOS 12.4 21F79 darwin-arm, locale en-IN)
    • Flutter version 3.0.5 at /Users/mahesh/Documents/flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision f1875d570e (5 days ago), 2022-07-13 11:24:16 -0700
    • Engine revision e85ea0e79c
    • Dart version 2.17.6
    • DevTools version 2.12.2

[✓] Android toolchain - develop for Android devices (Android SDK version 33.0.0-rc4)
    • Android SDK at /Users/mahesh/Library/Android/sdk
    • Platform android-32, build-tools 33.0.0-rc4
    • ANDROID_HOME = /Users/mahesh/Library/Android/sdk
    • Java binary at: /Applications/Android Studio.app/Contents/jre/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 11.0.12+0-b1504.28-7817840)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 13.2.1)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • CocoaPods version 1.11.2

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 2021.2)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.12+0-b1504.28-7817840)

[✓] IntelliJ IDEA Community Edition (version 2021.2.1)
    • IntelliJ at /Applications/IntelliJ IDEA CE.app
    • Flutter plugin version 61.2.4
    • Dart plugin version 212.5080.8

[✓] VS Code (version 1.67.2)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.44.0

[✓] Connected device (3 available)
    • sdk gphone arm64 (mobile) • emulator-5554 • android-arm64  • Android 11 (API 30) (emulator)
    • macOS (desktop)           • macos         • darwin-arm64   • macOS 12.4 21F79 darwin-arm
    • Chrome (web)              • chrome        • web-javascript • Google Chrome 103.0.5060.114

[✓] HTTP Host Availability
    • All required HTTP hosts are available

• No issues found!
```

```
[✓] Flutter (Channel master, 3.1.0-0.0.pre.1594, on macOS 12.4 21F79 darwin-arm, locale en-IN)
    • Flutter version 3.1.0-0.0.pre.1594 on channel master at /Users/mahesh/Documents/flutter_master
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 082ad11422 (2 days ago), 2022-07-11 11:40:07 +0200
    • Engine revision 2b5d5c3b2d
    • Dart version 2.18.0 (build 2.18.0-271.0.dev)
    • DevTools version 2.15.0

[✓] Android toolchain - develop for Android devices (Android SDK version 33.0.0-rc4)
    • Android SDK at /Users/mahesh/Library/Android/sdk
    • Platform android-32, build-tools 33.0.0-rc4
    • ANDROID_HOME = /Users/mahesh/Library/Android/sdk
    • Java binary at: /Applications/Android Studio.app/Contents/jre/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 11.0.12+0-b1504.28-7817840)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 13.2.1)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Build 13C100
    • CocoaPods version 1.11.2

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 2021.2)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.12+0-b1504.28-7817840)

[✓] IntelliJ IDEA Community Edition (version 2021.2.1)
    • IntelliJ at /Applications/IntelliJ IDEA CE.app
    • Flutter plugin version 61.2.4
    • Dart plugin version 212.5080.8

[✓] VS Code (version 1.67.2)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.44.0

[✓] Connected device (3 available)
    • Redmi K20 Pro (mobile) • 192.168.1.2:5555 • android-arm64  • Android 11 (API 30)
    • macOS (desktop)        • macos            • darwin-arm64   • macOS 12.4 21F79 darwin-arm
    • Chrome (web)           • chrome           • web-javascript • Google Chrome 103.0.5060.114

[✓] HTTP Host Availability
    • All required HTTP hosts are available

• No issues found!
```
</details>

**Comment by Burchard36** [comment]
> Hi @Burchard36, Thanks for filing the issue. Please ensure you are using the latest version of the `google_sign_in` plugin. I did verify on Android and it seems to return the serverAuthcode.
> 
> <img alt="image" width="790" src="https://user-images.githubusercontent.com/31410839/179529582-3a9ba633-4135-4b4b-b709-8077507e5810.png">
> 
> This is a duplicate of a recently fixed issue #57712 (closed)
> 
> flutter doctor -v (mac)

Hello the issue is I am using the latest version, which is why when i read through that post earlier it didnt quite make sense to me:

This is my pubspec.yaml (I have multiple other oath libs cause I was trying others after i could get this one working, it was the first on i tried on a fresh flutter project, removing them later tonight)
![image](https://user-images.githubusercontent.com/48583030/179635596-d7e82595-df11-4080-9e27-697084570855.png)
